### PR TITLE
don't use a block when #fetching with a default value

### PR DESCRIPTION
**Pull request by dunn**
When we aren't manipulating the value fetched, it's clearer to use the
default-value parameter instead of passing a block.

### CODE REVIEW

File: app/controllers/about_controller.rb [hunk 1]
```diff
@@ -17,9 +17,9 @@ def more
     @extended_description = Setting.site_extended_description
     @contact_account      = Account.find_local(Setting.site_contact_username)
     @contact_email        = Setting.site_contact_email
-    @user_count           = Rails.cache.fetch('user_count')            { User.count }
```
Comment (nightpool) [inline app/controllers/about_controller.rb]: This defeats the point of using a cache, though?

**Comment by nightpool** [comment]
In the case that the default value is expensive to compute, this is a performance regression.

**Comment by dunn** [comment]
Oh, I didn't realize there was a performance difference between `{ }.fetch(:a) { "default" }` and `{ }.fetch(:a, "default")`.  If that's the reason for this style, feel free to close this PR>

**Comment by Gargron** [comment]
Blocks are evaluated when they are explicitly called, unlike inline statements. Sorry!

**Comment by nightpool** [comment]
@Gargron the ENV ones could probably still use the change?

**Comment by nightpool** [comment]
There's no reason for `ENV.fetch('SMTP_FROM_ADDRESS') { 'notifications@localhost' }` vs `ENV.fetch('SMTP_FROM_ADDRESS', 'notifications@localhost')`, as far as I know.

**Pull request closed without merging**